File: MultiLanguageSandbox/src/test/deps/swift/6.swift
func countElementsWithStrictBounds(_ nums: [Int]) -> Int {
    guard !nums.isEmpty else { return 0 }
    let minVal = nums.min()!
    let maxVal = nums.max()!
    
    if minVal == maxVal {
        return 0
    }
    
    var count = 0
    for num in nums {
        if num > minVal && num < maxVal {
            count += 1
        }
    }
    return count
}


func checkCountElementsWithStrictBounds(_ countElementsWithStrictBounds: ([Int]) -> Int) {
assert(countElementsWithStrictBounds([2, 1, 3]) == 1)
assert(countElementsWithStrictBounds([1, 2, 3]) == 1)
assert(countElementsWithStrictBounds([1, 1, 1]) == 0)
assert(countElementsWithStrictBounds([3, 3, 5, 2, 2, 7, 4]) == 4)
assert(countElementsWithStrictBounds([10, 5, 9, 1, 7, 6, 8]) == 5)
assert(countElementsWithStrictBounds([-2, -1, 0, 1, 2]) == 3)
assert(countElementsWithStrictBounds([]) == 0)
}

checkCountElementsWithStrictBounds(countElementsWithStrictBounds)

File: MultiLanguageSandbox/src/test/deps/elixir/6.exs
defmodule MajorityElementFinder do
def find_majority_element(nums) do
    # Initialize candidate and count
    {candidate, count} = Enum.reduce(nums, {nil, 0}, fn num, {current_candidate, current_count} ->
      if current_count == 0 do
        {num, 1}
      else
        if num == current_candidate do
          {current_candidate, current_count + 1}
        else
          {current_candidate, current_count - 1}
        end
      end
    end)

    # Verify if the candidate is indeed the majority element
    if Enum.count(nums, &(&1 == candidate)) > div(length(nums), 2) do
      candidate
    else
      nil
    end
  end
end

# Test cases


ExUnit.start()
defmodule MajorityElementFinderTest do
use ExUnit.Case

test "finds the majority element" do
assert MajorityElementFinder.find_majority_element([3, 2, 3]) == 3
assert MajorityElementFinder.find_majority_element([2, 2, 1, 1, 1, 2, 2]) == 2
end

test "handles single element array" do
assert MajorityElementFinder.find_majority_element([1]) == 1
end

test "handles array where all elements are the same" do
assert MajorityElementFinder.find_majority_element([2, 2, 2]) == 2
end
end